pkg/build/api/v1/types.go

```diff
@@ -163,6 +163,9 @@ type BuildSource struct {
 	// repository.
 	ContextDir string `json:"contextDir,omitempty" description:"specifies sub-directory where the source code for the application exists, allows for sources to be built from a directory other than the root of a repository"`
 
+	// DockerfileName is the name of the Dockerfile that will be used to build the Docker image.
+	DockerfileName string `json:"dockerfileName,omitempty" description:"specifies the name of the Dockerfile to use for building the Docker image"`
```

Comment:
Which input source?  Git and Binary?  Name of the Dockerfile you want to
build from is pretty clearly a strategy argument.  It is not shared with
other build types.  If I provide a Dockerfile and a GIT repo as input, the
strategy is the one who makes those decisions.

There's no real reason for it to be generic to source, since it's not
generic to all strategies.  Only things generic to all strategies should be
in source.

On Nov 24, 2015, at 3:48 PM, Ben Parees <notifications@github.com> wrote:

In pkg/build/api/v1/types.go
<https://github.com/openshift/origin/pull/5976#discussion_r45792515>:

> @@ -163,6 +163,9 @@ type BuildSource struct {
>  	// repository.
>  	ContextDir string `json:"contextDir,omitempty" description:"specifies sub-directory where the source code for the application exists, allows for sources to be built from a directory other than the root of a repository"`
>
> +	// DockerfileName is the name of the Dockerfile that will be used to build the Docker image.
> +	DockerfileName string `json:"dockerfileName,omitempty" description:"specifies the name of the Dockerfile to use for building the Docker image"`

I haven't looked carefully at the Dockerfile input feature lately, but why
would you need to provide a dockerfile name when you're going through that
flow? To ensure you overwrite any existing Dockerfile provided in the input
source?

if so i would say that makes an argument for providing the dockerfile name
in the input source.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/5976/files#r45792515>.